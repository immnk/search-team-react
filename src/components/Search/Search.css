.root {
  background: linear-gradient(0deg, #000000e0, #1111117d),
    url('https://cdn.dribbble.com/users/1626229/screenshots/14571364/media/c29cef6df5ebfeb98bd4455392abaf7d.jpg')
      no-repeat;
  background-size: cover;
  background-position: center;
}

.container {
  width: 720px;
  margin: 0 auto;
  padding: 10rem 2.4rem;

  @media screen and (max-width: 540px) {
    width: 100%;
  }
}

.header {
  font-family: 'Lobster', cursive;
  font-weight: 400;
  font-size: 3.2rem;
  margin-bottom: 1.2rem;
  text-align: center;
  color: #fff;
}

.radioInput {
  font-size: 1.6rem;
  display: inline-flex;
  align-items: center;

  input {
    height: 2.5rem;
    width: 2.5rem;
    margin-right: 0.5rem;
  }

  span {
    color: #fff;
  }
}

.searchInput {
  border: 1px solid #cbcddc80;
  border-radius: 0.4rem;
  display: flex;
  align-items: center;
  margin: 1.4rem;
  box-shadow: 0 1px 3px #ffffff1f, 0 1px 2px #ffffff3d;
  transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
  background: #fff;

  input {
    border: none;
    display: block;
    flex: 1;
    font-size: 2.4rem;
    font-weight: 100;
    padding: 1.5rem 1.5rem 1.5rem 0;

    &:focus {
      outline: none;
    }
  }

  &:focus-within {
    box-shadow: rgb(255 255 255 / 12%) 0 1px 6px, rgb(255 255 255 / 12%) 0 1px 12px;
  }
}

.searchIcon {
  color: #4d4f59b3;
  margin: 0 1.4rem;
}

.categories {
  text-align: center;
  margin: 1.2rem 0;

  label:not(:last-child) {
    margin-right: 1.6rem;
  }
}
