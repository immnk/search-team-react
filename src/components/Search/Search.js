import React, { useState } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { connect } from 'redux-zero/react';
import ReactGA from 'react-ga';

import actions from 'store/search/actions';

import css from './Search.css';

const Search = ({ className, handleSearchInput, handleCategoryChange }) => {
  const [category, setCategory] = useState('name');

  function onTextChange(event) {
    const { value } = event.currentTarget;
    ReactGA.event({
      category: 'Search',
      action: 'Typing Search',
      label: value
    });
    handleSearchInput(event);
  }

  function onCategoryChange(event) {
    const { value } = event.currentTarget;
    ReactGA.event({
      category: 'Search',
      action: 'Category change',
      label: value
    });
    setCategory(value);
    handleCategoryChange(value);
  }

  return (
    <section className={classNames(css.root, className)}>
      <div className={css.container}>
        <h3 className={css.header}>Know your users</h3>
        <label htmlFor="search" className={css.searchInput}>
          <span className={classNames('material-icons', css.searchIcon)}>person_search</span>
          <input
            id="search"
            type="search"
            placeholder="Search user here..."
            autoComplete="off"
            autoCorrect="off"
            autoCapitalize="none"
            spellCheck="false"
            onChange={onTextChange}
          />
        </label>

        <section className={css.categories}>
          <label htmlFor="name" className={css.radioInput}>
            <input
              id="name"
              type="radio"
              name="category"
              value="name"
              checked={category === 'name'}
              onChange={onCategoryChange}
            />
            <span>Name</span>
          </label>
          <label htmlFor="email" className={css.radioInput}>
            <input
              id="email"
              type="radio"
              name="category"
              value="email"
              checked={category === 'email'}
              onChange={onCategoryChange}
            />
            <span>e-mail</span>
          </label>
          <label htmlFor="username" className={css.radioInput}>
            <input
              id="username"
              type="radio"
              name="category"
              value="username"
              checked={category === 'username'}
              onChange={onCategoryChange}
            />
            <span>Username</span>
          </label>
          <label htmlFor="company" className={css.radioInput}>
            <input
              id="company"
              type="radio"
              name="category"
              value="company.name"
              checked={category === 'company.name'}
              onChange={onCategoryChange}
            />
            <span>Company</span>
          </label>
        </section>
      </div>
    </section>
  );
};

Search.defaultProps = {
  className: undefined
};

Search.propTypes = {
  className: PropTypes.string,
  handleSearchInput: PropTypes.func.isRequired,
  handleCategoryChange: PropTypes.func.isRequired
};

export default connect(null, actions)(Search);
