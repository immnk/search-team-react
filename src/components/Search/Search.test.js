import React from 'react';
import { shallow } from 'enzyme';

import Search from './Search';

const renderSearch = (props) => {
  const defaultProps = {
    className: 'my-classname'
  };

  return <Search {...defaultProps} {...props} />;
};

const shallowRenderSearch = (props) => shallow(renderSearch(props));

describe('Test Search', () => {
  it('render app', () => {
    const actual = shallowRenderSearch();
    expect(actual).toMatchSnapshot();
  });
});
