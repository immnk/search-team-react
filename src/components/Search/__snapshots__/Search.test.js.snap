// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Search render app 1`] = `
<Connect
  actions={[Function]}
  className="my-classname"
  mapToProps={null}
>
  <Component />
</Connect>
`;
